classes/runner.php

```diff
@@ -844,6 +844,27 @@ public function addExtension(extension $extension, extension\configuration $conf
 		return $this;
 	}
 
+	public function getExtension($classname)
+	{
+		$extensions = array();
+		foreach ($this->getExtensions() as $extension)
+		{
+			if (get_class($extension) != $classname)
+			{
+				continue;
+			}
+
+			$extensions[] = $extension;
+		}
+
+		if (1 != count($extensions))
```

Comment:
I think we should improve the `addExtension` method and move this check there: atoum should refuse to load an extension twice.

The `addExtension` already checks for an already existing extension instance but it should also check if an extension of the same class is already loaded.

Doing this, we could make the code in `getExtension` simpler: return early in the foreach loop and throw if we exit the loop without any matching extension.